proxy-node-gateway/src/test/java/uk/gov/ida/notification/resources/EidasAuthnRequestResourceV2Test.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.ida.notification.resources;
+
+import com.google.common.collect.Lists;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ida.notification.EidasSamlParserService;
+import uk.gov.ida.notification.SamlFormViewBuilder;
+import uk.gov.ida.notification.VSPService;
+import uk.gov.ida.notification.dto.EidasSamlParserRequest;
+import uk.gov.ida.notification.dto.EidasSamlParserResponse;
+import uk.gov.ida.notification.dto.VSPAuthnRequestResponse;
+
+import javax.servlet.http.HttpSession;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static uk.gov.ida.notification.resources.EidasAuthnRequestResourceV2.SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_KEY;
+import static uk.gov.ida.notification.resources.EidasAuthnRequestResourceV2.SESSION_KEY_EIDAS_DESTINATION;
+import static uk.gov.ida.notification.resources.EidasAuthnRequestResourceV2.SESSION_KEY_EIDAS_REQUEST_ID;
+import static uk.gov.ida.notification.resources.EidasAuthnRequestResourceV2.SESSION_KEY_HUB_REQUEST_ID;
+import static uk.gov.ida.notification.resources.EidasAuthnRequestResourceV2.SUBMIT_BUTTON_TEXT;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EidasAuthnRequestResourceV2Test {
+
+    @InjectMocks
+    private EidasAuthnRequestResourceV2 resource;
+
+    @Mock
+    private EidasSamlParserService eidasSamlParserService;
+
+    @Mock
+    private VSPService vspService;
+
+    @Mock
+    private SamlFormViewBuilder samlFormViewBuilder;
+
+    @Mock
+    private HttpSession session;
+
+    @Mock
+    private EidasSamlParserResponse eidasSamlParserResponse;
+
+    @Mock
+    private VSPAuthnRequestResponse vspResponse;
+
+    @Mock
+    private Handler logHandler;
+
+    @Captor
+    private ArgumentCaptor<LogRecord> captorLoggingEvent;
+
+    @Captor
+    private ArgumentCaptor<EidasSamlParserRequest> captorEidasSamlParserRequest;
+
+    @Before
+    public void setup() {
```

Comment:
Do we need the teardown for this to remove that logger?

Reply:
Will add a teardown to remove mock handler from logger